newIDE/app/src/PropertiesEditor/PropertiesMapToSchema.js

```diff
@@ -242,4 +249,50 @@ const propertiesMapToSchema = (
   }));
 };
 
+const exponents = ['⁰', '¹', '²', '³', '⁴', '⁵'];
+
+export const getMeasurementUnitShortLabel = (
+  measurementUnit: gdMeasurementUnit
+): string => {
+  return mapFor(0, measurementUnit.getElementsCount(), i => {
+    const baseUnit = measurementUnit.getElementBaseUnit(i);
+    const power = measurementUnit.getElementPower(i);
+    const absPower = Math.abs(power);
+    const showPower = power < 0 || (absPower > 1 && absPower < 6);
+    return (
+      baseUnit.getSymbol() +
+      (power < 0 ? '⁻' : '') +
+      (showPower ? exponents[absPower] : '')
+    );
+  }).join(' · ');
+};
+
+export const getMeasurementUnitDescription = (
+  measurementUnit: gdMeasurementUnit
+): string => {
+  let wasNegativePower = false;
+  return (
+    '### ' +
+    measurementUnit.getLabel() +
+    '\n' +
+    measurementUnit.getDescription() +
+    '\n\n_in ' +
+    mapFor(0, measurementUnit.getElementsCount(), i => {
+      const baseUnit = measurementUnit.getElementBaseUnit(i);
+      const power = measurementUnit.getElementPower(i);
+      const absPower = Math.abs(power);
+      const signedUnit = (power < 0 ? 'per ' : '') + baseUnit.getLabel();
+      const separator = power < 0 ? ', ' : ' ';
+      const firstSeparator = i === 0 ? '' : wasNegativePower ? separator : ' ';
+      const poweredUnit =
+        firstSeparator +
+        signedUnit +
+        (absPower > 1 ? (separator + signedUnit).repeat(absPower - 1) : '');
+      wasNegativePower = power < 0;
+      return poweredUnit;
+    }).join('') +
```

Comment:
I wonder if it's a bad idea to generate the worded unit because it will be hard to translate ("per" is a bit alone).
Should we add a property in MeasurementUnit for it instead?

Reply:
I did this.